src/api/transactions.js

```diff
@@ -1,5 +1,94 @@
+const e = require('express');
 const express = require('express');
+const { Transaction } = require('../db/models');
 
 const router = express.Router();
 
+/**
+ * Get transactions grouped by date, month, or year between two intervals
+ * Sample query: /api/transactions/spikes?period=6
+ */
+router.get('/spikes', async (req, res, next) => {
+  try {
+    if (!req.user) {
+      return res.status(401).json({ error: 'Please log in' });
+    }
+
+    let { period } = req.query;
+    
+    // Spending X month ago where X by default is 1 year ago
+    const endDate = new Date();
+    const startDate = new Date();
+    startDate.setMonth(startDate.getMonth() - period);
```

Comment:
If the period value was not given or was improperly formatted, the API route would return a 500 instead of properly returning an error response.